crates/task-impls/src/quorum_proposal/handlers.rs

```diff
@@ -97,9 +104,78 @@ pub struct ProposalDependencyHandle<TYPES: NodeType, V: Versions> {
 
     /// The node's id
     pub id: u64,
+
+    /// The time this view started
+    pub view_start_time: Instant,
+
+    /// The higest_qc we've seen at the start of this task
+    pub highest_qc: QuorumCertificate<TYPES>,
 }
 
 impl<TYPES: NodeType, V: Versions> ProposalDependencyHandle<TYPES, V> {
+    /// Return the next HighQC we get from the event stream
+    async fn wait_for_qc_event(
+        &self,
+        rx: &mut Receiver<Arc<HotShotEvent<TYPES>>>,
+    ) -> Option<QuorumCertificate<TYPES>> {
+        while let Ok(event) = rx.recv_direct().await {
+            if let HotShotEvent::HighQcRecv(qc, _sender) = event.as_ref() {
+                if qc
+                    .is_valid_cert(
+                        self.quorum_membership.as_ref(),
+                        TYPES::Epoch::new(0),
```

Comment:
@lukaszrzasik Need to get a proper epoch here, I know we can get this from consensus, but need to do this properly like we discussed maybe it's trivial if the epoch is on the QC